src/Microsoft.AspNet.Mvc.ViewFeatures/Internal/ValidationHelpers.cs

```diff
@@ -23,7 +27,7 @@ public static string GetUserErrorMessageOrDefault(ModelError modelError, ModelSt
             }
 
             var attemptedValue = entry.AttemptedValue ?? "null";
-            return Resources.FormatCommon_ValueNotValidForProperty(attemptedValue);
+            return messageProvider.InvalidValueWithUnknownModelErrorAccessor(attemptedValue);
```

Comment:
Taking this one out represents a regression from MVC 5, where users can override this exact message.

If we're painting ourselves into a corner using `ModelMetadata` to carry these messages around, let's go back to one setting per message system-wide. (The per-property overrides supported in MVC 6 have no equivalent in earlier versions.) That is, just hang everything off `MvcOptions` and pass the `IModelBindingMessageProvider` into `[Try]AddModelError()` in addition to the `ModelMetadata` (with similar changes elsewhere).